server/s3_delete_multiple_objects_action.cc

```diff
@@ -89,6 +89,7 @@ void S3DeleteMultipleObjectsAction::setup_steps() {
   s3_log(S3_LOG_DEBUG, request_id, "Setting up the action\n");
   ACTION_TASK_ADD(S3DeleteMultipleObjectsAction::validate_request, this);
   ACTION_TASK_ADD(S3DeleteMultipleObjectsAction::fetch_objects_info, this);
+  ACTION_TASK_ADD(S3DeleteMultipleObjectsAction::save_bucket_counters, this);
```

Comment:
Saving counters must happen in the same loop where we remove objects' metadata.  Otherwise different scenarios happen:

1) iteration 1 - remove 30 objects; iteration 2 - remove 30 objects; iteration 3 InternalError -- counters NOT updated.
2) iteration 1 - remove 30 objects, iteration 2 ... iteration 55 of 60 - POWER OFF.  Counters not updated.

Reply:
Done